src/DigitalSubscriberAppBanner.tsx

```diff
@@ -283,19 +286,45 @@ export const storeIcon = css`
     }
 `;
 
+const withTryCatchBehaviour = (fn: () => void): void => {
+    try {
+        fn();
+    } catch (e) {}
+};
+
 export const DigitalSubscriberAppBanner: React.FC<Props> = ({
-    onButtonClick,
+    logButtonClickWithBraze,
+    submitComponentEvent,
     brazeMessageProps: { header, body },
 }: Props) => {
     const [showBanner, setShowBanner] = useState(true);
 
     const onCloseClick = (
         evt: React.MouseEvent<HTMLButtonElement, MouseEvent>,
-        buttonId: number,
+        internalButtonId: number,
     ): void => {
         evt.preventDefault();
-        onButtonClick(buttonId);
+
         setShowBanner(false);
+
+        withTryCatchBehaviour(() => {
+            // Braze displays button id from 1, but internal representation is numbered from 0
+            // This means that the Button ID in Braze and Ophan will be the same
+            const externalButtonId = internalButtonId + 1;
+            submitComponentEvent({
+                component: {
+                    // TODO: Fix this, the types need updating:
+                    componentType: 'RETENTION_ENGAGEMENT_BANNER' as OphanComponentType,
+                    id: 'DigitalSubscriberAppBanner',
```

Comment:
We've a point of fragility here, ensuring that this string stays in sync with the mappings. Can we put this hard coded string in one place to make ourselves a bit more resilient to changes in future?

Reply:
Ah great spot. Yep I'd meant to come back to this.

Reply:
I've split apart the component name from the implementation of the component in 7bf30d52f4180c6b45dc56142b59714c9e6c1fe9 which allows us to use the component name here.